tests/unit/layers/models/layersManager.spec.ts

```diff
@@ -0,0 +1,717 @@
+      mergeTilesTasker
+    );
+  });
+
+  describe('createLayer', () => {
+    it('should create "New" job type with "Split-Tiles" task type successfully', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1,2-3' });
+      setValue('ingestionTilesSplittingTiles.tasksBatchSize', 2);
+      const testData: IngestionParams = {
+        fileNames: ['test.tif'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+
+      getHighestLayerVersionMock.mockResolvedValue(undefined);
+      mapExistsMock.mockResolvedValue(false);
+      catalogExistsMock.mockResolvedValue(false);
+      fileValidatorValidateExistsMock.mockResolvedValue(true);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+      findJobsMock.mockResolvedValue([]);
+      createLayerJobMock.mockResolvedValue('testJobId');
+      createSplitTilesTasksMock.mockResolvedValue(undefined);
+
+      const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
+      layersManager = new LayersManager(
+        configMock,
+        jsLogger({ enabled: false }),
+        zoomLevelCalculator,
+        jobManagerClientMock,
+        catalogClientMock,
+        mapPublisherClientMock,
+        fileValidatorMock,
+        splitTilesTaskerMock,
+        mergeTilesTasker
+      );
+
+      await layersManager.createLayer(testData, managerCallbackUrl);
+      expect(getHighestLayerVersionMock).toHaveBeenCalledTimes(1);
+      expect(fileValidatorValidateExistsMock).toHaveBeenCalledTimes(1);
+      expect(findJobsMock).toHaveBeenCalledTimes(1);
+      expect(createSplitTilesTasksMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('should create "Update" job type with "Merge-Tiles" task type successfully when includes only GPKG files', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1,2-3' });
+      setValue('ingestionTilesSplittingTiles.tasksBatchSize', 2);
+      const testData: IngestionParams = {
+        fileNames: ['test.gpkg'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+
+      const getGridSpy = jest.spyOn(SQLiteClient.prototype, 'getGrid');
+      getGridSpy.mockReturnValue(Grid.TWO_ON_ONE);
+      getHighestLayerVersionMock.mockResolvedValue(2.0);
+      fileValidatorValidateExistsMock.mockResolvedValue(true);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+      mapExistsMock.mockResolvedValue(true);
+      findJobsMock.mockResolvedValue([]);
+      validateGpkgFilesMock.mockReturnValue(true);
+      createLayerJobMock.mockResolvedValue('testJobId');
+      createMergeTilesTasksMock.mockResolvedValue(undefined);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
```

Comment:
why the remarks?

Reply:
moved into before each test instad of create new instance every test manualy